usaspending_api/agency/v2/views/federal_account_list.py

```diff
@@ -20,15 +23,20 @@ class FederalAccountList(PaginationMixin, AgencyBase):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.params_to_validate = ["fiscal_year", "filter"]
+        self.params_to_validate = ["fiscal_year", "filter", "bureau_slug"]
+
+    def format_results(self, file_a_response, file_b_response):
+        for row_a, row_b in itertools.product(file_a_response, file_b_response):
```

Comment:
Why are you doing a cartesian product here? It seems like overkill when you only care about where rows match.

Reply:
Diving into this question ... 
@aguest-kc what determines the granularity of results getting returned by this API endpoint? 

- Is each record built and returned at the granularity of 1 record per: Agency? Federal Account? TAS? Bureau? Or is it variable -- and if so how?
- Should a result be returned for every File A record that matches the API filter params, and then augmented with totals from File B for the same criteria?
- Or is a result returned for every File B record that matches the API filter params, and then augmented with budget info from File A for the same criteria? 
- Or is a result returned for records EITHER File A OR File B records that match API filter params, and then the result is filled in with File A or File B data where it can be found?

Reply:
Here is an old PR from Wil that did something similar without the need to loop over every possible combination of results: https://github.com/fedspendingtransparency/usaspending-api/pull/3346/files#diff-87b27ed6231b4e97bfa0370021d6677ba778af56e5f1d0022ec1e9d7abe34f27R78

Reply:
Using SQL as an example, your `itertools.product()` approach is a CROSS JOIN which is typically not very performant or at least drops in performance very quickly as data grows. Wil instead uses an approach more inline with a FULL OUTER JOIN. Screenshot is added here for a visual representation of the two. 

![Screen Shot 2022-10-11 at 1 13 01 PM](https://user-images.githubusercontent.com/40359517/195189226-c24927be-8fa0-4e99-9676-00a256851569.png)

Reply:
I pushed an update that replaces `itertools.product` with an approach similar to the one from Wil's PR:

```python
# Combine File A and B responses
        combined_list_dict = {}

        for row in file_b_response:
            combined_list_dict[row['treasury_account__tas_rendering_label']] = row

        for row in file_a_response:
            # Append the total_budgetary_resources_amount_cpe value to the dict with the appropriate tas_rendering_label for this row
            if row['treasury_account_identifier__tas_rendering_label'] in combined_list_dict:
                combined_list_dict[row['treasury_account_identifier__tas_rendering_label']].update(
                    {'total_budgetary_resources': row['total_budgetary_resources_amount_cpe']}
                )
```